VerilogCodingStyle.md

```diff
@@ -1900,6 +1900,28 @@ separate combinational (`always_comb`) block. Ideally, sequential blocks should
 contain only a register instantiation, with perhaps a load enable or an
 increment.
 
+***All clock signals should be generated using blocking assignment even
+for clock dividers.***
+
+See #44 for details.
+
+&#x1f44d;
+```systemverilog {.good}
+// only for test bench code
+logic clk;
+initial begin
+  clk <= 1'b0;
+  forever #5 clk = ~clk;              // blocking assignment
+end
+
+// for both synthesizable and test bench code
```

Comment:
No, delays absolutely do not belong in synthesizable code. Clock dividers are to be provided by primitives--Delays are not synthesizable.

Reply:
Actually, I think your response makes this confusing again.

`#delay` is not being used for the clock divider implementation here; it's being used for the source clock _generation_. So let's stop making this about the `#delay` statement. It's still forbidden in synthesizable code, and that's irrelevant to your PR here.

Reply:
> No, delays absolutely do not belong in synthesizable code. Clock dividers are to be provided by primitives--Delays are not synthesizable.

Synthesis tools just ignore the #-delays, and it works fine on the example code for the clock divider.
But simulation tools require the #-delays when non-blocking assignment used.

I've seen the such kind of clock divider primitives many times.